src/auth/auth.service.ts

```diff
@@ -0,0 +1,73 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/prisma.service';
+import { RegisterDto } from './dto/register.dto';
+import { User } from '@prisma/client';
+import { hash, compare } from 'bcrypt';
+import { JwtService } from '@nestjs/jwt';
+import { LoginDto } from './dto/login.dto';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private prismaService: PrismaService,
+    private jwtService: JwtService,
+  ) {}
+
+  async register(userData: RegisterDto): Promise<User> {
+    // Check if user already exists
+    const user = await this.prismaService.user.findUnique({
+      where: {
+        email: userData.email,
+      },
+    });
+    if (user) {
+      throw new HttpException(
+        { message: 'This email is already registered' },
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+    const hashPassword = await hash(userData.password, 10);
+    const res = await this.prismaService.user.create({
+      data: { ...userData, password: hashPassword },
+    });
+
+    return res;
+  }
+
+  async login(userData: LoginDto): Promise<any> {
+    // Check if user exists
+    const user = await this.prismaService.user.findUnique({
+      where: {
+        email: userData.email,
+      },
+    });
+    if (!user) {
+      throw new HttpException(
+        { message: 'Invalid email or password' },
+        HttpStatus.UNAUTHORIZED,
+      );
+    }
+    // Check if password is correct
```

Comment:
and here.

nôm na là mấy chỗ này logic đang khá đơn giản,
ko cần comment detail đến mức đó